src/index.js

```diff
@@ -126,30 +127,58 @@ class Node extends EventEmitter {
     this.peerInfo.multiaddrs.replace(maOld, maNew)
 
     const multiaddrs = this.peerInfo.multiaddrs.toArray()
-    transports.forEach((transport) => {
-      if (transport.filter(multiaddrs).length > 0) {
-        this.switch.transport.add(
-          transport.tag || transport.constructor.name, transport)
-      } else if (WebSockets.isWebSockets(transport)) {
-        // TODO find a cleaner way to signal that a transport is always
-        // used for dialing, even if no listener
-        ws = transport
+
+    this._modules.transport.forEach((Transport) => {
+      let t
+
+      if (typeof Transport === 'function') {
+        t = new Transport()
+      } else {
+        t = Transport
       }
+
+      if (t.filter(multiaddrs).length > 0) {
+        this._switch.transport.add(t.tag || t.constructor.name, t)
+      } else if (WebSockets.isWebSockets(t)) {
+        // TODO find a cleaner way to signal that a transport is always used
+        // for dialing, even if no listener
+        ws = t
+      }
+      this._transport.push(t)
     })
 
     series([
-      (cb) => this.switch.start(cb),
+      (cb) => this._switch.start(cb),
       (cb) => {
         if (ws) {
           // always add dialing on websockets
-          this.switch.transport.add(ws.tag || ws.constructor.name, ws)
+          this._switch.transport.add(ws.tag || ws.constructor.name, ws)
         }
 
         // all transports need to be setup before discover starts
-        if (this.modules.discovery) {
-          return each(this.modules.discovery, (d, cb) => d.start(cb), cb)
+        if (this._modules.peerDiscovery && this._config.peerDiscovery) {
+          each(this._modules.peerDiscovery, (D, cb) => {
```

Comment:
It would be nice for the each callback to be named something else since we're touching this code. Being `cb` for `each` and for `series` makes this harder to follow than it needs to be.